google-summer-of-code/GSOC-2023-Ideas.md

```diff
@@ -0,0 +1,391 @@
+- Managing the deployment of the model and the server-side development at scale to handle large-scale deployments and accommodate faster inferences.
+- Development of a scalable user interface that introduces new input and output parameters with minimal effort/change in the client-side code-base for ease of maintenance.
+
+### Technical skills
+
+Experience designing and implementing Web Applications while managing end-to-end client and server-side development is a must! Understanding of a popular Python web framework (FastAPI, Flask, Django) is needed. The choice of the UI library/tools is flexible and will be based on the consensus of the mentor and the mentee. Understanding how a complete UI stack is set up would help the project be finished smoothly. Knowledge of the formats in which large geographical images are stored/rendered is preferred but optional.
+
+### First steps
+
+Please get started by reading and understanding the problem statement and our approach in this [publically available case study](https://community.moja.global/case-studies/powering-deforestation-prediction-using-artificial-intelligence). Try to understand the case study, the use-cases and how you can build an application around it. Share your thoughts on the `#machine-learning` channel and start working on the project's rough architecture and system design.
+
+## Idea: Object-oriented refactor of FLINT.Cloud
+
+### Abstract
+
+Moja global's community runs earth system simulations to track the flow of carbon in the forestry and land sectors. Models are calibrated to account for climate, forest types, disturbances (like fires, hurricanes, and so on), and land-use transitions. FLINT.Cloud is a cloud deployment framework for rapid deployment of FLINT implementations and offering FLINT as a demonstration service. It currently provides various models & simulations that allow an easy entry point for new users to evaluate the FLINT platform and provide a blueprint for new users to roll their own.
+
+Currently, the FLINT.Cloud offers two microservices: FLINT.Example REST API and GCBM REST API. The current implementation follows a functional programming paradigm that is difficult to scale. We have often faced problems implementing new API endpoints and following a consistent schema that would enable us to version our APIs and work on new features. We also wish to switch to an asynchronous, event-driven mode of executing the simulations to ensure a more efficient API layout and an increased processing velocity.
+
+The goal of this project is to implement an object-oriented refactor for FLINT.Cloud, including the addition of `Flask-RESTful` to extend the project modularity. The project should also involve the investigation and implementation of asynchronous task processing to make the carbon simulations more efficient. The current scope of work also entails implementing the unit, integration, and E2E tests to cover the new functional implementation and scale aspects.
```

Comment:
This is a great description - @SanjaySinghRajpoot also wanted to evaluate FastAPI, so perhaps we don't need to tie ourselves to FlaskRESTful at this stage?

Reply:
Sure, I will add this to the scope.